code/modules/mob/mob_movement.dm

```diff
@@ -496,24 +496,27 @@
 
 	var/turf/current_turf = get_turf(src)
 	var/turf/above_turf = SSmapping.get_turf_above(current_turf)
+	var/ventcrawling_mob = HAS_TRAIT(src, TRAIT_MOVE_VENTCRAWLING)
 
-	if(can_zFall(above_turf, 1, current_turf, DOWN)) //Will be fall down if we go up?
+	if(can_zFall(above_turf, 1, current_turf, DOWN) && !ventcrawling_mob) //Will be fall down if we go up?
 		to_chat(src, "<span class='notice'>You are not Superman.<span>")
 		return
 
-	if(zMove(UP, TRUE))
+	if(zMove(UP, TRUE, ventcrawling_mob))
 		to_chat(src, "<span class='notice'>You move upwards.</span>")
 
 ///Moves a mob down a z level
 /mob/verb/down()
 	set name = "Move Down"
 	set category = "IC"
 
-	if(zMove(DOWN, TRUE))
+	var/ventcrawling_mob = HAS_TRAIT(src, TRAIT_MOVE_VENTCRAWLING)
```

Comment:
what?

Reply:
to be able to move up and down while ventcrawling we need to check if the mob has that trait, so that when the proc checks for zMove down, it will pass that bool to the proc

Reply:
why is it called ventcrawling mob? it's not the ventcrawling mob

Reply:
Is more like a question "is the mob doing this movement vent crawling?"